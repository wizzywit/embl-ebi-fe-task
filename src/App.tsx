import React from 'react';
import Query from './components/Query/query.component';
import SearchResults from './components/SearchResults/searchResults.component';
import Search from './components/Search/search.component';
import useSearch from './hooks/Search.hook';

function App() {
  const {
    handleFetch,
    handleNext,
    handlePrev,
    currentPage,
    setQuery,
    query,
    queryString,
    data,
    previousMarks,
  } = useSearch();

  return (
    <main>
      <Search setQuery={setQuery} length={query.length} />
      <Query query={queryString} onClick={() => handleFetch()} />
      <SearchResults
        data={data}
        page={currentPage}
        onNext={handleNext}
        onPrev={handlePrev}
        isNext={data && !!data.nextCursorMark}
        isPrev={!!previousMarks.length}
      />
    </main>
  );
}

export default App;
