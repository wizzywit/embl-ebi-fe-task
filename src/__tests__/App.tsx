import { render, screen } from '@testing-library/react';
import App from '../App';
import axios from 'axios';
import GlobalStyle from '../globalStyle';
import React from 'react';

jest.mock('axios');
const mockedAxios = axios as jest.Mocked<typeof axios>;

test('should render the correct snapshot of the landing page', () => {
  mockedAxios.get.mockResolvedValueOnce([]);
  const { container } = render(
    <React.StrictMode>
      <GlobalStyle />
      <App />
    </React.StrictMode>,
  );
  expect(screen.getByText('Advanced search builder')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
