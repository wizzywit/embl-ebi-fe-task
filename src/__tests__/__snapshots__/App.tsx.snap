// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the correct snapshot of the landing page 1`] = `
<div>
  <main>
    <section
      class="sc-iBYQkv bLzXhO"
    >
      <div
        class="sc-ftTHYK ebeEzx"
      >
        <h1>
          Advanced search builder
        </h1>
        <div
          class="sc-pyfCe dWXjch"
        >
          <p>
            Add keyword or phrase
          </p>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M7 10l5 5 5-5H7z"
            />
          </svg>
        </div>
        <form
          class="sc-jrcTuL dcdnyE"
          role="search"
        >
          <label
            for="term"
          >
            Enter term or phrase:
          </label>
          <div
            class="form-group"
          >
            <div
              class="input-group"
            >
              <input
                aria-label="Enter term or phrase"
                aria-required="true"
                id="term"
                type="search"
                value=""
              />
            </div>
            <button
              class="sc-hLBbgP bDyqrH"
              disabled=""
              type="submit"
            >
              Add
            </button>
          </div>
        </form>
      </div>
    </section>
    <section
      class="sc-bcXHqe jkERTl"
    >
      <div
        class="sc-gswNZR qtAsH"
      >
        <p>
          Search query
        </p>
        <div
          class="query-group"
        >
          <div
            class="query"
            data-testid="query"
          />
          <button
            class="sc-hLBbgP bDyqrH"
            disabled=""
          >
            <div
              class="sc-dkrFOg gYfBNp"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0016 9.5 6.5 6.5 0 109.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
              <div>
                View search result
              </div>
            </div>
          </button>
        </div>
      </div>
    </section>
    <section
      class="sc-eDvSVe fgDUYp"
    >
      <div
        class="sc-jSUZER eruvAk"
      >
        <p>
          Search results
        </p>
        <p
          class="caption"
        >
          Enter a search term to view results
        </p>
      </div>
    </section>
  </main>
</div>
`;
